### ISSUE TITLE
github.com/tencentcloud/tencentcloud-sdk-go import issue on any tag after v1.7.5

### ISSUE THREAD

**Issue by szampardi** [hashicorp/vault#12742]
**Describe the bug**
`make bootstrap` fails with all Vault tags after v1.7.5

**To Reproduce**
```shell
+ git clone --branch v1.8.3 --depth 1 https://github.com/hashicorp/vault.git /opt/go/src/github.com/hashicorp/vault
Cloning into '/opt/go/src/github.com/hashicorp/vault'...
Note: switching to '73e85c3c21dfd1e835ded0053f08e3bd73a24ad6'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

Removing intermediate container 2b6e5ad38785
 ---> e502c45a12d2
Step 11/21 : WORKDIR	"${GOPATH}/src/github.com/hashicorp/vault"
 ---> Running in bba90516c59a
Removing intermediate container bba90516c59a
 ---> 52f3d0b7beb5
Step 12/21 : RUN	make bootstrap
 ---> Running in 9ecb27f64743
make bootstrap
+ make bootstrap
Installing/Updating github.com/elazarl/go-bindata-assetfs/...
Installing/Updating github.com/hashicorp/go-bindata/...
Installing/Updating github.com/mitchellh/gox
Installing/Updating golang.org/x/tools/cmd/goimports
go generate -tags tools tools/tools.go
go: github.com/hashicorp/go-discover@v0.0.0-20201029210230-738cb3105cd0 requires
	github.com/tencentcloud/tencentcloud-sdk-go@v3.0.83+incompatible: reading github.com/tencentcloud/tencentcloud-sdk-go/go.mod at revision v3.0.83: unknown revision v3.0.83
make: *** [Makefile:134: bootstrap] Error 1
```

**Expected behavior**
`make bootstrap` succeeds

**Environment:**
* Vault Server Version (retrieve with `vault status`): v1.8.3
* Vault CLI Version (retrieve with `vault version`):
* Server Operating System/Architecture: alpine 3.14 container with go1.17.1

Vault server configuration file(s):

```hcl
# Paste your Vault config here.
# Be sure to scrub any sensitive values
```

**Additional context**

**Comment by hsimon-hashicorp** [comment]
Hello. Go 1.17 will be supported in Vault 1.9, but you will not be able to build any previous versions with it. Thanks!

**Comment by szampardi** [comment]
thank you for the quick response, I wasn't even thinking about trying older versions of go yet 👍🏼

**Comment by szampardi** [comment]
@hsimon-hashicorp I am getting the same with go 1.16.8

**Comment by hsimon-hashicorp** [comment]
This looks like it might be a dependency issue that was recently fixed - https://github.com/hashicorp/vault/pull/12404. Take a look and let me know if that helps. Thanks!

**Comment by szampardi** [comment]
I don't know what to say, it looks like a few tags from https://github.com/TencentCloud/tencentcloud-sdk-go/tags have vanished. I am unable to compile any Vault tag after v1.7.4.

**Comment by szampardi** [comment]
```shell
+ git clone --branch main --depth 1 https://github.com/hashicorp/vault.git /opt/go/src/github.com/hashicorp/vault
Cloning into '/opt/go/src/github.com/hashicorp/vault'...
Updating files: 100% (4663/4663), done.
Removing intermediate container 2a42cfd08112
 ---> 4c8830a6bc63
Step 11/21 : WORKDIR	"${GOPATH}/src/github.com/hashicorp/vault"
 ---> Running in c49b8f96a800
Removing intermediate container c49b8f96a800
 ---> ca9a2c56f8e9
Step 12/21 : RUN	make bootstrap
 ---> Running in f7e98dc88a5f
make bootstrap
+ make bootstrap
Installing/Updating github.com/mitchellh/gox
Installing/Updating golang.org/x/tools/cmd/goimports
go generate -tags tools tools/tools.go
go: github.com/tencentcloud/tencentcloud-sdk-go@v3.0.171+incompatible: reading github.com/tencentcloud/tencentcloud-sdk-go/go.mod at revision v3.0.171: unknown revision v3.0.171
make: *** [Makefile:132: bootstrap] Error 1
The command '/bin/bash -evxo pipefail -c make bootstrap' returned a non-zero code: 2
```

**Comment by szampardi** [comment]
that's hard to believe considering I am building in a fresh docker container (https://github.com/szampardi/vault-docker/blob/master/stage1.Dockerfile), there is no chance for any cache at all.

Potentially the engineering team is who should be cleaning local mod cache, [my previous statement about the imported module](https://github.com/hashicorp/vault/issues/12742#issuecomment-934808738) remains still true, look for yourself: in the main branch go.mod you import `github.com/tencentcloud/tencentcloud-sdk-go v3.0.171+incompatible` (https://github.com/hashicorp/vault/blob/main/go.mod#L172), but tag does not exist (anymore?) https://github.com/TencentCloud/tencentcloud-sdk-go/releases/tag/v3.0.171 !

**Comment by szampardi** [comment]
@hsimon-hashicorp @calvn please note that this is still a problem with the last release v1.8.5

```shell
+ make bootstrap
[0mInstalling/Updating github.com/elazarl/go-bindata-assetfs/...
Installing/Updating github.com/hashicorp/go-bindata/...
Installing/Updating github.com/mitchellh/gox
Installing/Updating golang.org/x/tools/cmd/goimports
go generate -tags tools tools/tools.go
[91mgo: github.com/hashicorp/go-discover@v0.0.0-20201029210230-738cb3105cd0 requires
	github.com/tencentcloud/tencentcloud-sdk-go@v3.0.83+incompatible: reading github.com/tencentcloud/tencentcloud-sdk-go/go.mod at revision v3.0.83: unknown revision v3.0.83
[0m[91mmake: *** [Makefile:134: bootstrap] Error 1
[0mThe command '/bin/bash -evxo pipefail -c make bootstrap; 	cd ui && npx browserslist@latest --update-db' returned a non-zero code: 2
```

**Comment by jeffwidman** [comment]
@szampardi looks like the fix didn't land in the `1.8.x` line, but instead will ship as part of `1.9.0`...
